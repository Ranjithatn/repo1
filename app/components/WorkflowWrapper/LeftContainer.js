import React, { Fragment } from "react";

function LeftContainer({ formatMessage, heading, content, extendedContent, contentPreview }) {
  return (
    <Fragment>
      {heading && <div className="header">{heading}</div>}
      {content && <div className="dark-container">{content}</div>}
      {contentPreview && <div className="biometrics--preview-screen">{contentPreview}</div>}
      { extendedContent && <div className="extendedContent">{extendedContent}</div>}
    </Fragment>
  );
}

export default LeftContainer;
