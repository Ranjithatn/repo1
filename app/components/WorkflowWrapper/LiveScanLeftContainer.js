import React, { Fragment } from "react";

function LeftContainer({ formatMessage, heading, content, annotation,mugshot }) {
  return (
    <Fragment>
      <div className="header">{heading}</div>
      <div className="dark-container">{content}</div>    
      <div className="mugshot">{mugshot}</div>
      <div className="annotations">{annotation}</div>
      {/*<div className="mugshot">{mugshot}</div>*/}
    </Fragment>
  );
}

export default LeftContainer;
