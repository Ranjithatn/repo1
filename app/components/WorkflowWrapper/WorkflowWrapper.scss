@import "../../colors.scss";

.Workflow-Wrapper {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom:0;
  flex-grow: 1;
  display: grid;
  // grid-template-areas: "left top top top top"
  //                      "left main main main main"
  //                      "left main main main main"
  //                      "left main main main main"
  //                      "foot foot foot foot foot";
  grid-template-areas: "left top top top top"
                       "left main main main main"
                       "left main main main main"
                       "left main main main main"
                       "left main main main main";
  grid-template-columns: 250px 1fr 1fr 1fr 1fr;
  grid-template-rows: 50px 1fr 1fr 1fr 50px;
  .top {
    grid-area: top;
    background: $dark-gray;
    display: flex;
    align-items: center;
  }
  .left {
    grid-area: left;
    background: $light-gray;
    display: flex; // flex;
    flex-direction: column;
  }
  .main {
    grid-area: main;
    background-color: $dark;
    overflow: auto;
  }
  .foot {
    grid-area: foot;
    background: $light-gray;
    border: 1px solid $dark-gray;
  }
}

.Workflow-Wrapper .left {
  .header {
    display: flex;
    align-items: center;
    padding: 0 .75rem 0 .75rem;
    min-height: 80px;
    font-size: 26px;
  }
  .dark-container {
    // padding: 5px;
    display: flex;
    background: $dark;
    color: $white;
    font-size: 15px;
    min-height: 100px;
  }
  // .mugshot{
  //   border-bottom: 1px solid $dark;
  //   border-top: 1px solid $dark;
  // }
}


.VerifierWorkflow-Wrapper {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom:0;
  flex-grow: 1;
  display: grid;
  grid-template-areas: "left top top top top"
                       "left main main main main"
                       "left main main main main"
                       "left main main main main";
  grid-template-columns: 250px 1fr 1fr 1fr 1fr;
  grid-template-rows: 50px 1fr 1fr 1fr;
  overflow: auto;
  .top {
    grid-area: top;
    background: $dark-gray;
    display: flex;
    // align-items: center;
  }
  .left {
    grid-area: left;
    background: $light-gray;
    display: flex;
    flex-direction: column;
  }
  .main {
    grid-area: main;
    background-color: $dark;
    overflow: auto;
  }

  .main.full {
    grid-area: main;
    grid-column-start: left;
    background-color: $dark;
    overflow: auto;
  }
  .top.full {
    grid-area: top;
    grid-column-start: left;
    background: $dark-gray;
    display: flex;
    // align-items: center;
  }

}

.VerifierWorkflow-Wrapper .left {
  .header {
    display: flex;
    align-items: center;
    padding: 0 .75rem 0 .75rem;
    height: 80px;
    font-size: 26px;
  }
  .dark-container {
    padding: 5px;
    display: flex;
    background: $dark;
    color: $white;
    font-size: 15px;
    min-height: 100px;
  }
  .mugshot{
    border-bottom: 1px solid $dark;
    border-top: 1px solid $dark;
  }
}