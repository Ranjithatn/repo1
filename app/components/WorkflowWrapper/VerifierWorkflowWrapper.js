import React, { Fragment } from "react";
import { Switch, Route } from "react-router-dom";

import "./WorkflowWrapper.scss";

function VerifierWorkflowWrapper({ top, main, foot, left, className, panes }) {
  if ( ! panes ) { panes = {}; }
  return (
    <div style={{ position: "relative", flexGrow: "1" }}>
      <div className={"VerifierWorkflow-Wrapper" + (className ? " " + className : "")}>
        <div className={`top ${ ! panes.sidebar ? 'full' : '' }`}>{top}</div>
        { panes.sidebar &&
          <div className="left">{left}</div>
        }
        <div id="VerifierWorkflowWrapper-main" className={`main ${ ! panes.sidebar ? 'full' : '' }`}>
          {main}
        </div>
      </div>
    </div>
  );
}

export default VerifierWorkflowWrapper;
