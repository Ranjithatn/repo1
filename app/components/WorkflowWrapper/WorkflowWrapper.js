import React, { Fragment } from "react";
import { Switch, Route } from "react-router-dom";

import "./WorkflowWrapper.scss";

function WorkflowWrapper({ top, main, foot, left, className }) {
  return (
    <div style={{ position: "relative", flexGrow: "1" }}>
      <div className={"Workflow-Wrapper" + (className ? " " + className : "")}>
        <div className="top">{top}</div>
        <div className="left">{left}</div>
        <div id="WorkflowWrapper-main" className="main">
          {main}
        </div>
        {/* <div className="foot">{foot}</div> */}
      </div>
    </div>
  );
}

export default WorkflowWrapper;
