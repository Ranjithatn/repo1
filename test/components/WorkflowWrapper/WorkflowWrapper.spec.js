import React from "react";
import toJson from "enzyme-to-json";
import WorkflowWrapper from "../../../app/components/WorkflowWrapper/WorkflowWrapper";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<VerifierWorkflowWrapper />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <WorkflowWrapper  />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
