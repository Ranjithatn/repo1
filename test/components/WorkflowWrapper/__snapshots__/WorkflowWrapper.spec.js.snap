// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifierWorkflowWrapper /> should render without crashing 1`] = `
<WorkflowWrapper>
  <div
    style={
      Object {
        "flexGrow": "1",
        "position": "relative",
      }
    }
  >
    <div
      className="Workflow-Wrapper"
    >
      <div
        className="top"
      />
      <div
        className="left"
      />
      <div
        className="main"
        id="WorkflowWrapper-main"
      />
    </div>
  </div>
</WorkflowWrapper>
`;
